imap/mailbox.c

```diff
@@ -5728,19 +5728,20 @@ static int _mailbox_index_repack(struct mailbox *mailbox,
                 trecord->internal_flags = copyrecord.internal_flags;
                 ptrarray_append(records, trecord);
             }
-
-            // update the CRCs
-            repack->crcs.basic ^= crc_basic(&repack->newmailbox, &oldrecord);
-            repack->crcs.basic ^= crc_basic(&repack->newmailbox, &copyrecord);
-            repack->crcs.annot ^= crc_virtannot(&repack->newmailbox, &oldrecord);
-            repack->crcs.annot ^= crc_virtannot(&repack->newmailbox, &copyrecord);
-
-            // preload cache so we don't load it on newmailbox
-            r = mailbox_cacherecord(mailbox, &copyrecord);
-            if (r) goto done;
+            int dirty = oldrecord.internaldate.tv_sec != copyrecord.internaldate.tv_sec
+                     || oldrecord.internaldate.tv_nsec != copyrecord.internaldate.tv_nsec;
+
```

Comment:
Couldn't this also do struct->struct comparison like 
https://github.com/cyrusimap/cyrus-imapd/pull/5539/commits/2af6374c1914ab0744889052ba7fca802ab35bd2

Reply:
Actually, I think the mistake was removing the nsec check on reconstruct.  I removed it for bugs but I think they're all solved, I'll add it back.

You can't just == a struct.  You can copy it with =, but comparison would have to be `memcmp(a, b, sizeof(struct))` which is just fugly.  At least in non-C++.